### PR TITLE
Use default analyzer loader in Roslyn benchmarks

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to dotnet/roslyn#67604

Fixing the benchmark implementation to work with recently released Roslyn (4.3.0).

For the moment I deleted the custom analyzer loader. If .NET Native can't use this loader then ideally we could just fix that on the Roslyn side since presumably we need to be able to load analyzers in order to run the compiler on .NET Native.

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/benchmarks/real-world/Roslyn/StageBenchmarks.cs [hunk 1]
```diff
@@ -138,7 +133,8 @@ public Stream SerializeMetadata()
                 new SimpleEmitStreamProvider(_peStream),
                 metadataPEStreamProvider: null,
                 pdbStreamProvider: null,
-                testSymWriterFactory: null,
+                rebuildData: null,
+                testSymWriterFactory: null, // error CS8323: Named argument 'testSymWriterFactory' is used out-of-position but is followed by an unnamed argument [CompilerBenchmarks]csharp(CS8323)
```
Comment (RikkiGibson) [inline src/benchmarks/real-world/Roslyn/StageBenchmarks.cs:137]:
```suggestion
                testSymWriterFactory: null,
```

**Comment by RikkiGibson** [comment]
@dotnet/roslyn-compiler @dotnet/dotnet-perf-admin for review.

**Review by chsienki — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Review by LoopedBard3 — APPROVED** [review]
LGTM

**Comment by LoopedBard3** [comment]
@RikkiGibson Is this good to be merged?

**Comment by RikkiGibson** [comment]
Yes, thank you!

**Comment by kunalspathak** [comment]
@RikkiGibson - after this update, we saw some regressions on both x64 and arm64. Do you know why?

image

**Comment by RikkiGibson** [comment]
Moving from 3.8 to 4.3 brings in 2 years of additional work on the compiler. I think any major changes that happened in that timeframe could have done it.

Perhaps we should compare perf traces from before/after this PR? @dotnet/roslyn-compiler

**Comment by kunalspathak** [comment]
Here is another one from https://pvscmdupload.blob.core.windows.net/reports/allTestHistory%2f8.0_x64_Windows%2010.0.18362_RunKind%3droslyn%2fCompilerBenchmarks.StageBenchmarks.GetDiagnosticsWithAnalyzers.html

image

and from https://pvscmdupload.blob.core.windows.net/reports/allTestHistory%2f8.0_x64_Windows%2010.0.18362_RunKind%3droslyn%2fCompilerBenchmarks.StageBenchmarks.SerializeMetadata.html

image

**Comment by RikkiGibson** [comment]
Do the perf machines take ETL traces or anything like that for these scenarios? Otherwise, I imagine we will just have to pull down the before/after versions of the performance repo locally, take a trace, and compare?

**Comment by kunalspathak** [comment]
@LoopedBard3  can you help @RikkiGibson in getting ETL traces, if we are not already doing it?

**Comment by cincuranet** [comment]
@RikkiGibson ETLs are automatically taken for (most) auto filing issues. 

@kunalspathak was this auto filed?

**Comment by kunalspathak** [comment]
> @kunalspathak was this auto filed?

I am not sure, since we do not triage real world app's performance issues because of noise level. This was noticed from the newly created real-world benchmarks dashboard by @LoopedBard3 .

**Comment by jaredpar** [comment]
Is there a guide on how to use the website with the perf data? 

- When I select a range a popup says "double tap to zoom out". I've double tapped everywhere on the screen, nothing has changed. 
- When I click on markers on the graph I can "Set Baseline" and "Set Compare". Again though nothing changes. 
- If there are ETL files to look at they are _really_ hard to find.

**Comment by jaredpar** [comment]
Also how do we see what command was executed in both cases?

**Comment by LoopedBard3** [comment]
@RikkiGibson 
We have two other views depending on how in-depth you are looking to go looking into the data. The first is our PowerBI reports at [aka.ms/dotnetperfstatus](aka.ms/dotnetperfstatus). In this case, the Roslyn data is available in the RealWorldScenarios report found on the left side of the page. Once selected, use the Runkind, ShortTestName to select the test you want to view and then the other selectors to filter the exact run setup. Let me know if you are not able to access the PowerBI. 

We also have an Azure Data Explorer Dashboard here: [here](https://dataexplorer.azure.com/dashboards/bdd5e46a-6534-43b2-a280-1a32ead567cc?p-_startTime=60days&p-_endTime=now&p-_repo=v-dotnet%2Fcore-sdk&p-_branch=v-8.0&p-_queue=v-Windows.10.Amd64.19H1.Tiger.Perf&p-_buildArch=v-x64&p-_runKind=v-roslyn&p-_compMode=v-tiered&p-_otherConfigs=v-%7B%7D&p-_test=v-CompilerBenchmarks.RoslynApis.BuildAnalyzerConfigs#66e0f64f-9a80-4599-931d-467c577e2e63). From this page you can further explore the backing data by clicking the three dots and then explore query on the graph. For this you will need to join a security group, let me know if you want to try the ADX dashboard.

**Comment by jaredpar** [comment]
@LoopedBard3 

The link you posted seems incorrect. The text is correct but the link is wrong. This seems to be the correct link: https://aka.ms/dotnetperfstatus. I've looked at the PowerBI and it's unclear how to get it to show this specific regression. 

What would really help us is the ETL of the before and after. How do we get that?

Also how can we setup the PowerBI so that we can see the before after for `GetDiagnosticsWithAnalyzers` compared to `GetDiagnostics` for the same setup? That will help us understand what may be happening here. 

Further how do we see what actually ran here? What command are we actually measuring?

**Comment by kunalspathak** [comment]
You can try this command locally to repro.

```cmd
cd <repo>\src\benchmarks\real-world\Roslyn
dotnet build -c Release
dotnet <repo>\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll  --filter *BuildAnalyzerConfigs*
```

On my local machine, I did the above, reverted this PR and remeasured and I can repro the regression:

With PR:

|               Method |     Mean |    Error |   StdDev |   Median |      Min |      Max |     Gen0 |     Gen1 |    Gen2 | Allocated |
|--------------------- |---------:|---------:|---------:|---------:|---------:|---------:|---------:|---------:|--------:|----------:|
| BuildAnalyzerConfigs | 28.27 ms | 0.282 ms | 0.405 ms | 28.31 ms | 27.31 ms | 28.88 ms | 437.5000 | 375.0000 | 31.2500 |   2.65 MB |

Without PR:
|               Method |     Mean |    Error |   StdDev |   Median |      Min |      Max |     Gen0 |     Gen1 | Allocated |
|--------------------- |---------:|---------:|---------:|---------:|---------:|---------:|---------:|---------:|----------:|
| BuildAnalyzerConfigs | 26.95 ms | 0.262 ms | 0.291 ms | 26.96 ms | 26.48 ms | 27.47 ms | 343.7500 | 312.5000 |   2.14 MB |

With that, you should be able to gather the ETL traces to further investigate. Hope that helps!

**Comment by kunalspathak** [comment]
You can also change the filter to something else depending on what you want to investigate. You can see the full names of class by using:

```cmd
dotnetr <repo>\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll --filter * --list flat
CompilerBenchmarks.RoslynApis.BuildAnalyzerConfigs
CompilerBenchmarks.StageBenchmarks.CompileMethodsAndEmit
CompilerBenchmarks.StageBenchmarks.SerializeMetadata
CompilerBenchmarks.StageBenchmarks.GetDiagnostics
CompilerBenchmarks.StageBenchmarks.GetDiagnosticsWithAnalyzers
```

You do not need PowerBI or anything like that to repro this.

**Comment by LoopedBard3** [comment]
* Fixed the link, didn't know external links had to be prefixed with https://. 
* We don't get ETL files for every run we do unless there is an autofile issue so we need to run the ETL gathering manually. 
* For comparing the two tests in PowerBI, if you control click the selectors for the test names it will allow you to multi-select, and then you can filter based on the other selectors. 
* The actual test code is in this file: https://github.com/dotnet/performance/blob/main/src/benchmarks/real-world/Roslyn/StageBenchmarks.cs, with the supporting code being in that Roslyn folder.

**Comment by LoopedBard3** [comment]
Thank you @kunalspathak for getting that local run, let me see what else we may need to get ETL files generated!

**Comment by jaredpar** [comment]
@kunalspathak your local repro though shows a minimal change in performance, this bug says there is a 25% change.

**Comment by kunalspathak** [comment]
> @kunalspathak your local repro though shows a minimal change in performance, this bug says there is a 25% change.

True, it could vary depending on what other things were running on my local machine when I measured it, which might not be the case on perf machines.

**Comment by jaredpar** [comment]
@kunalspathak the issue isn't reproducing locally and the official measurements won't provide ETL files. How are we supposed to investigate this?

**Comment by kunalspathak** [comment]
> @kunalspathak the issue isn't reproducing locally and the official measurements won't provide ETL files. How are we supposed to investigate this?

It doesn't reproduce to the 25% extent, but locally it is 4%. Are you not able to reproduce this locally on your machine?

@LoopedBard3 - If you have access to the perf lab machine, can you see if it reproduces to the similar extent and if yes, share the ETL?

**Comment by LoopedBard3** [comment]
I have access to the lab machines so I will check to see if I can repro this regression and generate ETL files 👍.

**Comment by jaredpar** [comment]
@kunalspathak 

> You can try this command locally to repro.

This does not work. Apparently you must use .NET SDK 8.0 to build this repo but it can only run on .NET SDK 7.0. I spent a good 20 minutes this morning just working out how to get this to work.

**Comment by jaredpar** [comment]
> It doesn't reproduce to the 25% extent, but locally it is 4%. 

It's 4% on a 25ms benchmark. Further it's not an apples to apples comparison. The change @RikkiGibson made here to how assemblies are loaded is substantially different. The previous code just loaded assemblies directly into the current `AssemblyLoadContext`, the new change uses a separate `AssemblyLoadContext` with a lot of custom management on top of it.

**Comment by kunalspathak** [comment]
> This does not work. Apparently you must use .NET SDK 8.0 to build this repo but it can only run on .NET SDK 7.0

Oh yes, I typically use the `dotnet` present in `.dotnet` folder of the runtime repo. It is usually one of the preview versions of .net8. I use it to build and run it.

> with a lot of custom management on top of it.

If this will be the new baseline, I am fine with it as long as we know that it is indeed the cause of spike.

**Comment by JulieLeeMSFT** [comment]
Thanks @LoopedBard3. We will be waiting for the repro and etl files. 

> I have access to the lab machines so I will check to see if I can repro this regression and generate ETL files 👍.

**Comment by jaredpar** [comment]
@kunalspathak 

> Oh yes, I typically use the dotnet present in .dotnet folder of the runtime repo. It is usually one of the preview versions of .net8. I use it to build and run it.

The .net8 runtime alone though is not enough to run it. 

```cmd
> .dotnet/dotnet exec C:\Users\jaredpar\code\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll --filter BuildAnalyzerConfigs
You must install or update .NET to run this application.

App: C:\Users\jaredpar\code\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '7.0.0' (x64)
.NET location: C:\Users\jaredpar\code\performance\.dotnet\

The following frameworks were found:
  8.0.0-preview.1.23110.8 at [C:\Users\jaredpar\code\performance\.dotnet\shared\Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed
``` 

If you then attempt to use a .net7 runtime to run the code you get a different error because under the hood the benchmark re-compiles the application which requires .net8 

```cmd
🍿🐻 > dotnet exec C:\Users\jaredpar\code\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll --filter *BuildAnalyzerConfigs*
// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 1 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet  restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\jaredpar\code\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\8616fed0-2ef4-4474-9c24-1194b3a3b99b
// command took 2.11s and exited with 0
// start dotnet  build -c Release --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\jaredpar\code\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\8616fed0-2ef4-4474-9c24-1194b3a3b99b
// command took 1.47s and exited with 1
// start dotnet  build -c Release --no-restore --no-dependencies /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\jaredpar\code\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\8616fed0-2ef4-4474-9c24-1194b3a3b99b
// command took 3.02s and exited with 1
// ***** Done, took 00:00:06 (6.68 sec)   *****
// Found 1 benchmarks:
//   RoslynApis.BuildAnalyzerConfigs: Job-HYUGWZ(MaxRelativeError=0.01)

Setup power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: High performance)
// Build Error: Standard output:

 Standard error:
 MSBuild version 17.5.1+f6fdcf537 for .NET
C:\Users\jaredpar\.nuget\packages\microsoft.dotnet.xlifftasks\1.0.0-beta.23175.1\build\Microsoft.DotNet.XliffTasks.targets(114,5): error MSB4062: The "GatherXlf" task could not be loaded from the assembly C:\Users\jaredpar\.nuget\packages\microsoft.dotnet.xlifftasks\1.0.0-beta.23175.1\build\..\tools\net8.0\Microsoft.DotNet.XliffTasks.dll. Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Users\jaredpar\code\performance\src\harness\BenchmarkDotNet.Extensions\BenchmarkDotNet.Extensions.csproj]
C:\Users\jaredpar\.nuget\packages\microsoft.dotnet.xlifftasks\1.0.0-beta.23175.1\build\Microsoft.DotNet.XliffTasks.targets(114,5): error MSB4062: The "GatherXlf" task could not be loaded from the assembly C:\Users\jaredpar\.nuget\packages\microsoft.dotnet.xlifftasks\1.0.0-beta.23175.1\build\..\tools\net8.0\Microsoft.DotNet.XliffTasks.dll. Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Users\jaredpar\code\performance\src\tools\Reporting\Reporting\Reporting.csproj]
Build FAILED.
```

**Comment by kunalspathak** [comment]
> The .net8 runtime alone though is not enough to run it.

Here is my logs for build / run:

```
e:\git\performance\src\benchmarks\real-world\Roslyn>dotnetr build -c Release
E:\git\runtime\.dotnet
MSBuild version 17.6.0-preview-23174-01+e7de13307 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  Reporting -> e:\git\performance\artifacts\bin\Reporting\Release\netstandard2.0\Reporting.dll
  BenchmarkDotNet.Extensions -> e:\git\performance\artifacts\bin\BenchmarkDotNet.Extensions\Release\netstandard2.0\BenchmarkDotNet.Extensions.dll
  CompilerBenchmarks -> e:\git\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:13.56

e:\git\performance\src\benchmarks\real-world\Roslyn>dotnetr build -c Release
E:\git\runtime\.dotnet
MSBuild version 17.6.0-preview-23174-01+e7de13307 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  Reporting -> e:\git\performance\artifacts\bin\Reporting\Release\netstandard2.0\Reporting.dll
  BenchmarkDotNet.Extensions -> e:\git\performance\artifacts\bin\BenchmarkDotNet.Extensions\Release\netstandard2.0\BenchmarkDotNet.Extensions.dll
  CompilerBenchmarks -> e:\git\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.29

e:\git\performance\src\benchmarks\real-world\Roslyn>dotnetr e:\git\performance\artifacts\bin\CompilerBenchmarks\Release\net7.0\CompilerBenchmarks.dll --filter * --list flat
E:\git\runtime\.dotnet
CompilerBenchmarks.RoslynApis.BuildAnalyzerConfigs
CompilerBenchmarks.StageBenchmarks.CompileMethodsAndEmit
CompilerBenchmarks.StageBenchmarks.SerializeMetadata
CompilerBenchmarks.StageBenchmarks.GetDiagnostics
CompilerBenchmarks.StageBenchmarks.GetDiagnosticsWithAnalyzers
```


Perhaps I have lot of SDKs installed thats why I am not seeing any problems?

```
e:\git\performance\src\benchmarks\real-world\Roslyn>E:\git\runtime\.dotnet\dotnet.exe --info
.NET SDK:
 Version:   8.0.100-preview.3.23178.7
 Commit:    e300b0e1e6

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   E:\git\runtime\.dotnet\sdk\8.0.100-preview.3.23178.7\

.NET workloads installed:
There are no installed workloads to display.

Host:
  Version:      8.0.0-preview.3.23174.8
  Architecture: x64
  Commit:       47bad717bd

.NET SDKs installed:
  6.0.100 [E:\git\runtime\.dotnet\sdk]
  7.0.100-preview.3.22179.4 [E:\git\runtime\.dotnet\sdk]
  7.0.100-preview.5.22307.18 [E:\git\runtime\.dotnet\sdk]
  7.0.100-preview.7.22377.5 [E:\git\runtime\.dotnet\sdk]
  7.0.100-rc.1.22431.12 [E:\git\runtime\.dotnet\sdk]
  7.0.100 [E:\git\runtime\.dotnet\sdk]
  8.0.100-alpha.1.23061.8 [E:\git\runtime\.dotnet\sdk]
  8.0.100-preview.1.23115.2 [E:\git\runtime\.dotnet\sdk]
  8.0.100-preview.3.23178.7 [E:\git\runtime\.dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.3.22178.4 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.5.22303.8 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.7.22376.6 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-rc.1.22427.2 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-alpha.1.23058.7 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-preview.1.23112.2 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-preview.3.23177.8 [E:\git\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.3.22175.4 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.5.22301.12 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.7.22375.6 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-rc.1.22426.10 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-alpha.1.23057.5 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-alpha.1.23058.2 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-preview.1.23110.8 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-preview.3.23174.8 [E:\git\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.0 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.3.22177.1 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.5.22302.5 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.7.22377.1 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-rc.1.22427.1 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.0-alpha.1.23057.1 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.0-preview.1.23112.2 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.0-preview.3.23178.1 [E:\git\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

```

**Comment by jaredpar** [comment]
> Perhaps I have lot of SDKs installed thats why I am not seeing any problems?

That it mostly certainly why. At the same time though this repo is about performance so which runtime we use is important. Everyone needs to be using a consistent runtime otherwise we risk talking past each other because we're getting different results from different runtimes. As presently setup, basically by using the .NET SDK installed via the global.json file, the repository is non-functional. That means both

1. Hard to be confident we're all comparing the same data here. 
2. Hard to know how the actual lab runs are getting their data. It's clearly not using the runtime in the repo so which one is it using? How can I repo the lab data locally, etc ...

**Comment by kunalspathak** [comment]
> basically by using the .NET SDK installed via the global.json file, the repository is non-functional.

@LoopedBard3 - can we fix it? 

> Hard to know how the actual lab runs are getting their data

Let's see what @LoopedBard3 gathers for us.

**Comment by LoopedBard3** [comment]
Hey, I have done some investigation into this to see if I could repro the issue. While digging into the data I found a bug in the PowerBI reports that we have that caused this issue to look much worse than it actually was. After fixing the bug locally the difference comes back down to the 5ish% that Kunal was able to find. This 5% also correlates to the 2 other graphs that looked like regressions, as the super stable results make the 5% jump look massive. I will have the fix for the PowerBIs pushed by EOD Monday. @jaredpar With the big change not being the case and the difference only being 5% are you still looking for ETL files from the lab machine specifically? I added a small set of steps that can be used to generate the results locally, and it pulls down dotnet and what not.

Simple instructions for running the benchmarks and getting results like the lab:
Must have python installed
clone the performance repo
cd into the perf repo
`py -3 .\scripts\benchmarks_ci.py --csproj .\src\benchmarks\real-world\Roslyn\CompilerBenchmarks.csproj --incremental no --architecture x64 -f net8.0 --bdn-arguments="--anyCategories roslyn"`


If you want it to generate an ETW trace, add `-p ETW` to the bdn-arguments and run the command from an admin terminal. Let me know if these don't work. Also, if you also want to filter a benchmark, such as GetDiagnosticsWithAnalyzers, add `--filter *GetDiagnosticsWithAnalyzers*` to the --bdn-arguments string.

**Pull request merged**